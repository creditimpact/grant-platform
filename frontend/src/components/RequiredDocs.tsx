type Upload = { fields?: any };
type Item = {
  key: string;
  label: string;
  optional?: boolean;
  accept_mime?: string[];
  notes?: string;
  examples?: string[];
  uploads?: Upload[];
};

export default function RequiredDocs({ items }: { items: Item[] }) {
  return (
    <div className="space-y-3">
      {items.map((i) => (
        <div key={i.key} className="rounded-xl border p-4">
          <div className="font-medium">
            {i.label}{" "}
            {i.optional && (
              <span className="text-sm text-gray-500">(optional)</span>
            )}
          </div>
          {i.uploads && i.uploads.length ? (
            <ul className="text-sm mt-1 space-y-1">
              {i.uploads.slice(0, 3).map((u, idx) => {
                const f = u.fields || {};
                const parts = ["Payment Confirmation"];
                if (typeof f.payment_amount === "number") {
                  parts.push(
                    new Intl.NumberFormat("en-US", {
                      style: "currency",
                      currency: "USD",
                    }).format(f.payment_amount)
                  );
                }
                if (f.payment_date) parts.push(f.payment_date);
                if (f.confirmation_number)
                  parts.push(`#${f.confirmation_number}`);
                return <li key={idx}>{parts.join(" • ")}</li>;
              })}
            </ul>
          ) : (
            <>
              {i.accept_mime && (
                <div className="text-sm">
                  Accepted: {i.accept_mime.join(", ")}
                </div>
              )}
              {i.examples?.length ? (
                <div className="text-sm">
                  Examples: {i.examples.join(" • ")}
                </div>
              ) : null}
              {i.notes ? (
                <div className="text-sm text-gray-500">{i.notes}</div>
              ) : null}
            </>
          )}
        </div>
      ))}
    </div>
  );
}
