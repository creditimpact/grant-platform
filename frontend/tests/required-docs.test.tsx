import { render } from "@testing-library/react";
import RequiredDocs from "../src/components/RequiredDocs";

test("renders Tax_Payment_Receipt preview", () => {
  const items = [
    {
      key: "Tax_Payment_Receipt",
      label: "Tax Payment Receipt / Payment Confirmation",
      uploads: [
        {
          fields: {
            confirmation_number: "ABC12345",
            payment_amount: 1234.56,
            payment_date: "2023-04-15",
          },
        },
      ],
    },
  ];
  const { container } = render(<RequiredDocs items={items} />);
  expect(container).toMatchSnapshot();
});
