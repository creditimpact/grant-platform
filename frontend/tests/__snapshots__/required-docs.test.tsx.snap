// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tax_Payment_Receipt preview 1`] = `
<DocumentFragment>
  <div
    class="space-y-3"
  >
    <div
      class="rounded-xl border p-4"
    >
      <div
        class="font-medium"
      >
        Tax Payment Receipt / Payment Confirmation 
      </div>
      <ul
        class="text-sm mt-1 space-y-1"
      >
        <li>
          Payment Confirmation • $1,234.56 • 2023-04-15 • #ABC12345
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
